### DISCUSSION TITLE
Failed to decode request contents.

### DISCUSSION THREAD

**Discussion by cbzmqx** [w7corp/easywechat#2724] [Q&A] [answered]
当我配置企业微信接收事件服务器地址，需要响应企业微信解密后的明文消息内容，却始终出现Failed to decode request contents.的错误

**Comment by overtrue** [comment]
在哪里看到 “Failed to decode request contents“ 的?

**Reply by TheSaltwaterRoom** [reply]
微信公众号，基本配置
Failed to decode request contents.

**Reply by overtrue** [reply]
截图看看

**Reply by TheSaltwaterRoom** [reply]
问题我解决了，当我在微信公众号配置域名,点击提交时，微信会请求我的接口，
我当时调用了 $message = $server->getDecryptedMessage(); 然后报错
我改成了以下代码，就可以了
 ```php
$server->with(function ($message, \Closure $next) {
                // 你的自定义逻辑
                $this->logger->info('收到微信的消息', ['message' => $message]);
                return $next($message);
            });
```

**Comment by thissai** [comment, marked as answer]
遇到同样的问题：
```php
EasyWeChat\Kernel\Exceptions\BadRequestException: Failed to decode request contents. in
vendor/w7corp/easywechat/src/Kernel/Message.php:64<br />
Stack trace:<br />
#0 vendor/w7corp/easywechat/src/Kernel/Message.php(39): EasyWeChat\Kernel\Message::format('')<br />
#1 vendor/w7corp/easywechat/src/Work/Server.php(266): EasyWeChat\Kernel\Message::createFromRequest(Object(Nyholm\Psr7\ServerRequest))<br />
#2 vendor/w7corp/easywechat/src/Work/Server.php(60): EasyWeChat\Work\Server->getRequestMessage(Object(Nyholm\Psr7\ServerRequest))<br />
#3 app/controller/qywxController.php(49): EasyWeChat\Work\Server->serve()<br />
```
貌似是没有解析出 get 参数

**Reply by reatang** [reply]
微信还有get请求吗？麻烦提供一下是哪个接口

**Reply by thissai** [reply]
175.27.32.31 - - [22/Aug/2023:21:33:41 +0800] "GET /api/qywx/notify?msg_signature=d974e3fb04c3f3bd818c976ba37459e648305ab0&timestamp=1692711219&nonce=1692509890&echostr=18HKU8hrBQB2gnq5ILmkghqaNiNijZZN0Ezp6PVsCj0QLNm7%2BFLEumx80bhhvBaCf2mwM44Y9t3OD9I1skayYw%3D%3D HTTP/1.1" 500 2233 "-" "Mozilla/4.0"

配置接收消息的api地址的时候，微信会主动发一个get 请求

https://developer.work.weixin.qq.com/document/10514#%E8%AE%BE%E7%BD%AE%E6%8E%A5%E6%94%B6%E6%B6%88%E6%81%AF%E7%9A%84%E5%8F%82%E6%95%B0

**Comment by cbzmqx** [comment]
最新版本依然有这个问题

**Reply by thissai** [reply]
我后面这块没用easywechat，用 php 代码硬接的请求，可以搜搜 php 源码改改，其他部分还是用的 easywechat
